end_to_end_tests/interface.py

```diff
@@ -0,0 +1,103 @@
+# Copyright 2020 Google Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Interface for e2e tests."""
+
+import time
+import unittest
+import os
+import inspect
+import traceback
+from collections import Counter
+
+from timesketch_api_client.client import TimesketchApi
+
+# Default values based on Docker config.
+TEST_DATA_DIR = '/usr/local/src/timesketch/end_to_end_tests/test_data'
+HOST_URI = 'http://127.0.0.1'
+USERNAME = 'test'
+PASSWORD = 'test'
+
+
+class BaseEndToEndTest(object):
+    """Base class for end to end tests.
+
+    Attributes:
+        name: Test name
+        client: Instance of an API client
+        sketch: Instance of Sketch object
+        assertions: Instance of unittest.TestCase
+    """
+
+    NAME = 'name'
+
+    def __init__(self):
+        """Initialize the analyzer object."""
+        self.name = self.NAME
```

Comment:
any real need to do this? why not just use `self.NAME` when needed? are you planning on having the plugin have the ability to change the name dynamically?

Reply:
Left over from the analyzer interface. Removed